src/algebra/jordan/triple.lean

```diff
@@ -0,0 +1,215 @@
+/-
+Copyright (c) 2022 Christopher Hoskin. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Christopher Hoskin
+-/
+
+import algebra.lie.of_associative
+
+/-!
+# Jordan triples
+
+Let `A` be a module over a ring `R`. A triple product on `A` is a trilinear map
+$\{.\,.\,\}:A\times A\times A\mapsto A$ which is symmetric in the first and third variables. The
+module `A` is said to be a Jordan triple if, for any `a`, `b`, `c`, `d` and `e` in `A` the following
+Lebintz rule is satisfied:
+$$
+\{a\,b\,\{c\,d\,e\}\} = \{\{a\,b\,c\}\,d\,e\} - \{c\,\{b\,a\,d\}\,e\} + \{c\,d\,\{a\,b\,e\}\}
+$$
+A module `A` over a *-ring `R` is said to be a *-triple if it has a triple product linear and
+symmetric in the first and thrid variable and conjugate linear in the second variable. A *-triple
+satisfying the Lebintz rule is said to be a Jordan *-triple.
+
+As well as being of algebraic interest, Jordan *-triples arise naturally in mathematical physics,
+functional analysis and differential geometry. For more information about these connections the
+interested reader is referred to [alfsenshultz2003], [chu2012], [friedmanscarr2005],
+[iordanescu2003] and [upmeier1987].
+
+## Main results
+
+(None yet, we're just getting started.)
+
+## References
+
+* [Chu, Jordan Structures in Geometry and Analysis][chu2012]
+-/
+
+/-- Notation class for the triple product -/
+class has_tp (A : Type*) := (tp : A → A → A → A )
+
+notation ⦃a, b, c⦄ := has_tp.tp a b c
+
+/-- A triple product is a trilinear map, symmetric in the first and third variable -/
+class is_tp (A : Type*) [has_tp A] [has_add A] :=
+(comm : ∀ (a b c : A), ⦃a, b, c⦄ = ⦃c, b, a⦄)
+(ladd : ∀ (a₁ a₂ b c : A), ⦃(a₁+a₂), b, c⦄ = ⦃a₁, b, c⦄ + ⦃a₂, b, c⦄)
+(madd : ∀ (a b₁ b₂ c : A), ⦃a, (b₁+b₂), c⦄ = ⦃a, b₁, c⦄ + ⦃a, b₂, c⦄)
+
+/-- A Jordan triple product satisfies a Lebintz law -/
```

Comment:
Is this a typo for Leibniz?

Reply:
Yes, sorry - spelling not my strong point. Fixed.